server/src/main/java/org/opensearch/cluster/routing/allocation/AllocationService.java

```diff
@@ -548,6 +565,23 @@ private void allocateExistingUnassignedShards(RoutingAllocation allocation) {
             existingShardsAllocator.beforeAllocation(allocation);
         }
 
+        /*
+         Use batch mode if enabled and there is no custom allocator set for Allocation service
+         */
+        Boolean batchModeEnabled = EXISTING_SHARDS_ALLOCATOR_BATCH_MODE.get(settings);
+        if (batchModeEnabled && allocation.nodes().getMinNodeVersion().onOrAfter(Version.CURRENT) && existingShardsAllocators.size() == 2) {
```

Comment:
the check for ==2 here is for GatewayAllocator & BatchGatewayAllocator?

Reply:
Yes.